crates/ruff_cli/src/commands/format.rs

```diff
@@ -155,47 +193,50 @@ fn format_path(
 
     // Format the source.
     match format_source(source_kind, source_type, Some(path), settings)? {
-        FormattedSource::Formatted(formatted) => {
-            if mode.is_write() {
+        FormattedSource::Formatted {
+            unformatted,
+            formatted,
+        } => match mode {
+            FormatMode::Write => {
                 let mut writer = File::create(path).map_err(|err| {
                     FormatCommandError::Write(Some(path.to_path_buf()), err.into())
                 })?;
                 formatted
                     .write(&mut writer)
                     .map_err(|err| FormatCommandError::Write(Some(path.to_path_buf()), err))?;
+                Ok(FormatResult::Formatted)
             }
-            Ok(FormatResult::Formatted)
-        }
+            FormatMode::Check => Ok(FormatResult::Formatted),
+            FormatMode::Diff => Ok(FormatResult::Diff {
+                path: path.to_path_buf(),
```

Comment:
Can we make this unowned, by storing `&Path`? The lifetimes should work out, as far as I can tell.

Reply:
I missed that we already have the path on the outer struct that we can use